org.eclipse.jdt.ui/ui/org/eclipse/jdt/internal/ui/javaeditor/ClassFileEditorInputFactory.java

```diff
@@ -61,14 +73,25 @@ public IAdaptable createElement(IMemento memento) {
 				if (project != null) {
 					type= project.findType(type.getFullyQualifiedName());
 					if (type == null)
-						return null;
+						return new InternalClassFileEditorInput(cf);
```

Comment:
If one wants, here we can pass a boolean to Indicate that the type was not found on the project if that's useful for anything later on.